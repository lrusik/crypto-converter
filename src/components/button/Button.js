import React from 'react'
import "./button.css";

export default function Button({label}) {
   return (
      <button data-testid="button" id="button" className="main-button">
         {label}
      </button>
   )
}
