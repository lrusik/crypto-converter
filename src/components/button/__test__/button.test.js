import React from "react";
import ReactDOM from "react-dom";
import Button from "./../Button";

import renderer from "react-test-renderer";
import { render } from '@testing-library/react';
import '@testing-library/jest-dom'

it('Renders without crashing', () =>  {
   const div = document.createElement('div');
   ReactDOM.render(<Button />, div);
   ReactDOM.unmountComponentAtNode(div);
});

it('Renders button correctly', () => {
   const {getByTestId} = render(<Button label="click me" />);
   expect(getByTestId('button')).toHaveTextContent("click me");
});

it("matches snapshot", () => {
   const tree = renderer.create(<Button label="click me"></Button>).toJSON();
   expect(tree).toMatchSnapshot();
});
