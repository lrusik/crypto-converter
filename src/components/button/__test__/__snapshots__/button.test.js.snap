exports[`test matches snapshot 1`] = `
<button
  className="main-button"
  data-testid="button"
  id="button">
  click me
</button>
`;
